analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarSymbolAnalysisContext.cs

```diff
@@ -20,24 +20,24 @@
 
 namespace SonarAnalyzer;
 
-public sealed class SonarSymbolAnalysisContext : SonarAnalysisContextBase<SymbolAnalysisContext>
+public sealed class SonarSymbolAnalysisContext : SonarReportingContextBase<SymbolAnalysisContext>
 {
-    public override SyntaxTree Tree => Context.GetFirstSyntaxTree();
+    public override SyntaxTree Tree => Context.Symbol.Locations.Select(x => x.SourceTree).FirstOrDefault(x => x is not null);
```

Comment:
Would be great if we would take into consideration the PR-modified files and ensure a repeatable order:
- if Roslyn is not consistent with the order of location we will change the issue location from one analysis to the other confusing the user
- if a PR analysis is done and we pick an unmodified file the diagnostic will not be visible.
Both these cases can be annoying for the user.

Reply:
That's a good point. Also we might consistently choose generated file and that's not good. 

As this improvement is it's out of scope of this refactoring, I've created #6610 to do that.